packages/web/src/components/nav/desktop/NavAudio.tsx

```diff
@@ -121,4 +110,18 @@ const NavAudio = () => {
   )
 }
 
+const NavAudio = () => {
+  const userChallengesLoading = useSelector(getUserChallengesLoading)
```

Comment:
have some thoughts, will do a thorough pass

Reply:
Yeah. So there are quite a few places where we have selector logic normalizing things like this to empty values. I presume it was to avoid errors when attempting to render values which haven't loaded yet.
I was waiting on the cache/fetch refactoring to define a pattern before trying to address this more directly.

Ideally, I'd like to have _all_ fetchable items share the same interface where you have `data` and then metadata indicating the state of the fetch. So you can safely ignore the value of `data` until the fetch is in a terminal state.

The pattern I have here with the fetching and presentational components is IMO a clean way to deal with the issue. Normally, I'd do something like `return isLoading ? <NavAudioSkeleton /> : <NavAudio balance={balance} />`